code/modules/mob/living/carbon/human/human_mob.dm

```diff
@@ -2182,3 +2182,76 @@ Eyes need to have significantly high darksight to shine unless the mob has the X
 /mob/living/carbon/human/proc/apply_offstation_roles(source)
 	SIGNAL_HANDLER
 	mind.offstation_role = TRUE
+
+/mob/living/carbon/human/proc/expert_chef_knowledge()
+	set name = "Expert Chef Knowledge"
+	set desc = "Examine your surroundings, see what you can cook."
+	set category = "Martial Arts"
+
+	if(!HAS_TRAIT(mind, TRAIT_EXPERT_CHEF))
+		return
+	if(has_status_effect(STATUS_EXPERT_CHEF))
+		to_chat(src, "<span class='warning'>All this thinking makes your head hurt, wait a bit longer.</span>")
+		return
+	apply_status_effect(STATUS_EXPERT_CHEF)
+
+	var/list/possible_cookware = orange(1, src)
+
+	for(var/obj/item/storage/storage in possible_cookware)
+		possible_cookware += storage.contents
+		possible_cookware -= storage
+
+	shuffle_inplace(possible_cookware)
+
+
+	var/list/available_ingredients = list()
+	var/list/available_reagents = list()
+	for(var/obj/item/item in possible_cookware)
+		if(item.container_type == OPENCONTAINER && item.reagents)
+			for(var/datum/reagent/reagent in item.reagents.reagent_list)
+				if(!available_reagents[reagent.id])
+					available_reagents[reagent.id] = 0
+				available_reagents[reagent.id] += reagent.volume
+		else
+			if(!available_ingredients[item.type])
+				available_ingredients[item.type] = 0
+			available_ingredients[item.type] += 1
+
+	var/list/possible_recipes = list()
```

Comment:
You should probably also look for the basic cooking reagent recipes: dough, pastry dough, cheese, chocolate.

Knife and processing recipes would be nice, too, but are probably out of scope for this PR.